src/scripts.c

```diff
@@ -556,6 +559,36 @@ static int kiavc_lua_method_getversionstring(lua_State *s) {
 	return 1;
 }
 
+/* Require version as major, minor, patch */
+static int kiavc_lua_method_requireversion(lua_State *s) {
+	/* This method allows the Lua script to check whether the engine is of a compatible version (and quit if not) */
+	int n = lua_gettop(s), exp = 3;
+	if(n != exp) {
+		SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "[Lua] Wrong number of arguments in requireVersion: %d (expected %d)\n", n, exp);
+		kiavc_cb->quit();
+		return 0;
+	}
+	int major = luaL_checknumber(s, 1);
+	if(major != KIAVC_VERSION_MAJOR) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "[Lua] KIAVC major version number incompatible: %d (expected %d)\n", KIAVC_VERSION_MAJOR, major);
+        kiavc_cb->quit();
+        return 0;
+    }
+	int minor = luaL_checknumber(s, 2);
+	if(minor > KIAVC_VERSION_MINOR) {
```

Comment:
I get the SemVer reference you made, but I think it can cause ambiguity if someone is interested in a specific version. Maybe there could be different checks, where one checks if the version is exactly that, and another one if it is _at least_ that (which could loosen the check on the major version as well). We could even just keep the "exact version" check, and leave more flexible checks to the application (using the getters for the three version numbers that exist).

Reply:
I wouldn't want to loosen the major version in a minimum requirement, because that number is meant to signal breaking API changes that are not backwards-compatible.

My proposed solution is to have two formats, like you said, one being the exact one (now `requireEngineVersion`), but the other one being the SemVer one, which I renamed to `requireEngineCompatible`. See commits.